File: Project-Euler/Solutions-Simon/euler16/euler16.jl



num = parse(Int, ARGS[1])
power = parse(Int, ARGS[2])
sum = 0

#arr = [2]
arr = zeros(Int, 400)
arr[1] = 2
carry = 0

for i = 1 : power
  #push!(arr, 0)
  for j = 1 : length(arr) #Something else here?
    temp = arr[j] * 2
    if temp > 9
      #println(temp)
      arr[j] = temp % 10 + carry
      carry = 1
    else
      arr[j] = temp + carry
      carry = 0
    end
  end
end
#println(length(arr))

for i = 1 : length(arr)
  sum += arr[i]
  print(arr[i])
end

println("\n$sum")

File: Project-Euler/Solutions-Simon/euler16/euler16.java

public class euler16 {
  public static void main (String[] args) {
      int num = Integer.parseInt(args[0]);
      int power = Integer.parseInt(args[1]);

      int[] arr = new int[400];
      arr[0] = 2;
      int carry = 0;
      int sum = 0;

      for (int i = 1; i < 1000; i++) {
        for (int j = 0; j < arr.length; j++) {
          int temp = arr[j] * 2;
          if (temp > 9) {
            arr[j] = temp % 10 + carry;
            carry = 1;
          }
          else {
            arr[j] = temp + carry;
            carry = 0;
          }
        }
      }
      for (int i = 0; i < arr.length; i++) {
        sum += arr[i];
        System.out.print(arr[i]);
      }
      System.out.println("\n" + sum);
  }
}

